Celeste.Mod.mm/Patches/Monocle/EntityList.cs

```diff
@@ -115,5 +131,24 @@ static partial class MonoModRules {
             cursor.Next.Operand = hashRemoveOperand;
         }
 
+        public static void PatchEntityListAddAndRemove(ILContext context, CustomAttribute attrib) {
+            // insert the following code at the beginning of the method
+            // if (entity == null) throw new ArgumentNullException("entity")
+
+            TypeDefinition t_ArgumentNullException = MonoModRule.Modder.FindType("System.ArgumentNullException").Resolve();
+            MethodReference ctor_ArgumentNullException = MonoModRule.Modder.Module.ImportReference(t_ArgumentNullException.FindMethod("System.Void .ctor(System.String)"));
+
+            ILCursor cursor = new ILCursor(context);
+            ILLabel label = cursor.DefineLabel();
+            cursor.Emit(OpCodes.Ldarg_1);
+            cursor.Emit(OpCodes.Ldnull);
+            cursor.Emit(OpCodes.Ceq);
+            cursor.Emit(OpCodes.Brfalse_S, label);
```

Comment:
Not 100% sure if it's semantically the same but can't we replace this with:
```Csharp
cursor.Emit(OpCodes.Brtrue_S, label);
```

Reply:
I **Think** that if someone somehow passes `true` or non-zero c# boxes it so we won't throw false positives on the null check.

Reply:
i think they are the same, but use brture is more cleanner